src/rust/engine/src/nodes.rs

```diff
@@ -155,38 +155,39 @@ impl Select {
 
   pub fn new_from_edges(
     params: Params,
-    product: TypeId,
+    dependency_key: &DependencyKey<TypeId>,
     edges: &rule_graph::RuleEdges<Rule>,
   ) -> Select {
-    let dependency_key = selectors::DependencyKey::JustSelect(selectors::Select::new(product));
-    // TODO: Is it worth propagating an error here?
```

Comment:
Why remove the TODO?

Reply:
We've never encountered this panic in the wild, so the answer to the TODO's question is "no".